guides/creating-a-custom-form-field-control.md

```diff
@@ -177,19 +182,39 @@ constructor(
 ) { }
 ```
 
-Note that if your component implements `ControlValueAccessor`, it may already be set up to provide `NG_VALUE_ACCESSOR` (in the `providers` part of the component's decorator, or possibly in a module declaration). If so you may get a *cannot instantiate cyclic dependency* error.
+Note that if your component implements `ControlValueAccessor`, it may already be set up to provide
+`NG_VALUE_ACCESSOR` (in the `providers` part of the component's decorator, or possibly in a module
+declaration). If so you may get a *cannot instantiate cyclic dependency* error.
 
 To resolve this, remove the `NG_VALUE_ACCESSOR` provider and instead set the value accessor directly:
 
 ```ts
-constructor(
-  ...,
-  @Optional() @Self() public ngControl: NgControl,
-  ...,
-) {
-  // Setting the value accessor directly (instead of using
-  // the providers) to avoid running into a circular import.
-  if (this.ngControl != null) { this.ngControl.valueAccessor = this; }
+@Component({
+  ...
+  providers: [
+    ...
+    // Remove the this.
+    {
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: forwardRef(() => MatFormFieldControl),
+			multi: true,
```

Comment:
Is the indentation intentional here?

Reply:
Looks like I took this from somewhere that was indented with tabs, instead of spaces.

Reply:
Done.